cli/README.md

```diff
@@ -0,0 +1,206 @@
+
+### Command Line Tutorial
+
+First, set an environment variable for the Divvi Up account and an API token. The initial token will be created when an account is created.
+
+```
+export TOKEN=YOUR_TOKEN_HERE
+```
+
+Test the token by listing all accounts.
+
+```
+divviup --token $TOKEN account list
+```
+
+List the aggregators and identify the ID of both a leader and a helper.
+
+```
+divviup --token $TOKEN aggregator list
+```
+
+```
+  {
+    "id": "3650870b-56e6-4eac-8944-b7ca36569b33",
+    "role": "Either",
+    "name": "Divvi Up staging-dap-09-1",
+    "dap_url": "https://staging-dap-09-1.api.example.com/",
+    "api_url": "https://staging-dap-09-1.api.example.com/aggregator-api",
+    "is_first_party": true,
+    "vdafs": [
+      "Prio3Count",
+      "Prio3Sum",
+      "Prio3Histogram",
+      "Prio3SumVec"
+    ],
+    "query_types": [
+      "TimeInterval",
+      "FixedSize"
+    ],
+    "protocol": "DAP-09",
+    "features": [
+      "TokenHash",
+      "TimeBucketedFixedSize",
+      "UploadMetrics"
+    ]
+  },
+  {
+    "id": "96301951-c848-4a57-b4f5-32812e4db1be",
+    "account_id": null,
+    "created_at": "2024-03-21T22:47:15.467139Z",
+    "updated_at": "2024-04-18T17:33:30.439465Z",
+    "deleted_at": null,
+    "role": "Either",
+    "name": "Divvi Up staging-dap-09-2",
+    "dap_url": "https://staging-dap-09-2.api.example.com/",
+    "api_url": "https://staging-dap-09-2.api.example.com/aggregator-api",
+    "is_first_party": false,
+    "vdafs": [
+      "Prio3Count",
+      "Prio3Sum",
+      "Prio3Histogram",
+      "Prio3SumVec"
+    ],
+    "query_types": [
+      "TimeInterval",
+      "FixedSize"
+    ],
+    "protocol": "DAP-09",
+    "features": [
+      "TokenHash",
+      "UploadMetrics",
+      "TimeBucketedFixedSize"
+    ]
+  }
+```
+
+Set the two IDs into environment variables. NOTE: These IDs will vary based on your configuration.
+
+```
+export LEADERID=3650870b-56e6-4eac-8944-b7ca36569b33
+export HELPERID=96301951-c848-4a57-b4f5-32812e4db1be
+```
+
+Next, generate a collector-credential for the task. The collector credential will be used by the collector to export the aggregated statistics.
+
+```
+divviup --token $TOKEN collector-credential generate
+```
+
+```
+{
+  "id": "0a0f8ea8-b603-4416-b138-b7f217153bb7",
+  "hpke_config": {
+    "id": 144,
+    "kem_id": "X25519HkdfSha256",
+    "kdf_id": "HkdfSha256",
+    "aead_id": "Aes128Gcm",
+    "public_key": "V9IpdJxS91MHPiNTjwDk9DFS-5M_neVrPxlmvolmTTo"
+  },
+  "created_at": "2024-05-03T15:23:56.624726Z",
+  "deleted_at": null,
+  "updated_at": "2024-05-03T15:23:56.624727Z",
+  "name": "collector-credential-144",
+  "token_hash": "VItYJdAyWYIvooe8GzkGnVTvaMkvWc9G-eiwxudfWww",
+  "token": "A6JDAYPiYDXmNyh-OpYXGw"
+}
+
+New collector credential generated. Copy and paste the following text into a file or your password manager:
+{
+  "aead": "AesGcm128",
+  "id": 144,
+  "kdf": "Sha256",
+  "kem": "X25519HkdfSha256",
+  "private_key": "SECRETS",
```

Comment:
FWIW the token is secret (and arguably its token_hash) too, so we may as well redact that too.

Reply:
I think we do this so that the subscriber has a chance to back up the private key.

However, I agree it isn't great to print private key material to stdout by default. An advanced subscriber might run this tool in a suitably-secure production environment (e.g. a VM running in the subscriber's could environment, perhaps) & parse the private key; but this would be hindered by the fact that this output is a mixture of human-readable & JSON output.

Perhaps we shouldn't print these credentials (by default?) and should instead export them to a file? This is IMO no worse for the "subscriber runs this tool manually from their workstation" perspective, and would make it easier to do the actually-secure thing of running this tool in a production environment.

In any case, this isn't a problem to be solved with this PR.

Reply:
We do have functionality on that subcommand to dump to a file, but it's not the default option. I'll look around for a bit to see what the status quo is for "command that generates sensitive information" (ssh-add stands out in my mind as always defaulting to a file, maybe it should behave like ssh-add).

Reply:
I think we might want to make some improvements to certain `divviup` subcommands or other tools, but not in the scope of this PR, so I'll resolve this conversation.